mephisto/data_model/task.py

```diff
@@ -380,7 +386,12 @@ def sync_completion_status(self) -> None:
             for status in AssignmentState.incomplete():
                 if statuses[status] > 0:
                     has_incomplete = True
-            if not has_incomplete:
+            if (
+                not has_incomplete
+                and self.assignments_generator_done is None
+                or not has_incomplete
+                and self.assignments_generator_done
```

Comment:
This conditional can be made simpler, what are you trying to capture in this logic?

Reply:
I'm trying to let both `TaskRun` (the one that's run by the `Operator` and the one that is independent) to access this line.

Reply:
I can't follow this logic very clearly though as written considering it has a number of `and`s and `or`s interlinked, with two of the same `not has_incomplete` call. Going to take steps to simplify.
```python
if not has_incomplete and self.assignments_generator_done is None or not has_incomplete and self.assignments_generator_done: # The logic in here is very difficult to parse given the many statements and no divisions
if (not has_incomplete and self.assignments_generator_done is None) or (not has_incomplete and self.assignments_generator_done): # My guess is that this is your intended behavior, as python gives precedence to and over or, but this isn't easily seen without additional division.
if (not has_incomplete) and (self.assignments_generator_done is None or self.assignments_generator_done): # Given the above, it's possible to pull out the not has_incomplete term
if (not has_incomplete) and (self.assignments_generator_done is not False): # The latter term of the above can also be simplified, as we know assignments_generator_done only has 3 possible states.
if not has_incomplete and self.assignments_generator_done is not False:
```